salt/_grains/caasp_nodename.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/python
+
+import salt.grains.core
+
+def caasp_nodename():
+    """
+    Override core.hostname grain to provide lowercased items
+    localhost, fqdn, host, domain, nodename
+    """
+    grains = salt.grains.core.hostname()
+    return dict(
+        domain = grains['domain'].lower(),
+        fqdn = grains['fqdn'].lower(),
+        host = grains['host'].lower(),
+        localhost = grains['localhost'].lower(),
+        nodename = grains['host'].lower()
```

Comment:
Why not use the original `nodename` grain here?

Reply:
Side note: salt in our current version gets this information by calling to `platform.uname()`: https://github.com/saltstack/salt/blob/3e844ed1df13fea1b8fbc4d5b9d5b4f130743364/salt/grains/core.py#L1187-L1188

Reply:
@ereslibre do we expect the hostname we are getting from uname to be different from the host grain? If so - this sounds like a bug.

Reply:
@FedericoCeratto blame systemd and wicked :) Systemd has 3 different hostnames - static, transient and pretty. These no guarantee these line up, and no guarantee these match uname, /etc/hostname, etc. Wicked also updates some, but not all, of these when for example a DHCP hostname is provided.

Reply:
TBH, I'd prefer not to do any assumptions here and just transform the very same grain we've been working with. Worst case, you cannot access to the original grain because it's still being populated I think the safest bet here would be us calling `platform.uname()` to retrieve the `nodename`, if we cannot read the original grain itself and transform it (that for me would be the ideal case -- since it's something we can forget about for future salt versions).

Reply:
I agree with @ereslibre : we should just do a `nodename = grains['nodename'].lower()`